src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/schema/charts/metric.ts

```diff
@@ -184,36 +176,54 @@ export const metricStateSchema = schema.allOf([
-      schema.allOf([metricStatePrimaryMetricOptionsSchema, staticOperationDefinitionSchema]),
-      schema.allOf([metricStatePrimaryMetricOptionsSchema, uniqueCountMetricOperationSchema]),
+      // oneOf allows only 12 items
+      // so break down metrics based on the type: field-based, reference-based, formula-like
+      schema.oneOf([
+        schema.allOf([metricStatePrimaryMetricOptionsSchema, countMetricOperationSchema]),
+        schema.allOf([metricStatePrimaryMetricOptionsSchema, uniqueCountMetricOperationSchema]),
+        schema.allOf([metricStatePrimaryMetricOptionsSchema, metricOperationSchema]),
+        schema.allOf([metricStatePrimaryMetricOptionsSchema, sumMetricOperationSchema]),
+        schema.allOf([metricStatePrimaryMetricOptionsSchema, lastValueOperationSchema]),
+        schema.allOf([metricStatePrimaryMetricOptionsSchema, percentileOperationSchema]),
+        schema.allOf([metricStatePrimaryMetricOptionsSchema, percentileRanksOperationSchema]),
+      ]),
+      schema.oneOf([
+        schema.allOf([metricStatePrimaryMetricOptionsSchema, differencesOperationSchema]),
+        schema.allOf([metricStatePrimaryMetricOptionsSchema, movingAverageOperationSchema]),
+        schema.allOf([metricStatePrimaryMetricOptionsSchema, cumulativeSumOperationSchema]),
+        schema.allOf([metricStatePrimaryMetricOptionsSchema, counterRateOperationSchema]),
+      ]),
+      schema.oneOf([
+        schema.allOf([metricStatePrimaryMetricOptionsSchema, staticOperationDefinitionSchema]),
+        schema.allOf([metricStatePrimaryMetricOptionsSchema, formulaOperationDefinitionSchema]),
+      ]),
     ]),
     /**
      * Secondary value configuration, must define operation.
      */
     secondary_metric: schema.maybe(
       schema.oneOf([
-        schema.allOf([metricStateSecondaryMetricOptionsSchema, counterRateOperationSchema]),
-        schema.allOf([metricStateSecondaryMetricOptionsSchema, countMetricOperationSchema]),
-        schema.allOf([metricStateSecondaryMetricOptionsSchema, cumulativeSumOperationSchema]),
-        schema.allOf([metricStateSecondaryMetricOptionsSchema, differencesOperationSchema]),
-        schema.allOf([metricStateSecondaryMetricOptionsSchema, formulaOperationDefinitionSchema]),
-        schema.allOf([metricStateSecondaryMetricOptionsSchema, lastValueOperationSchema]),
-        schema.allOf([metricStateSecondaryMetricOptionsSchema, metricOperationSchema]),
-        schema.allOf([metricStateSecondaryMetricOptionsSchema, movingAverageOperationSchema]),
-        schema.allOf([metricStateSecondaryMetricOptionsSchema, percentileOperationSchema]),
-        schema.allOf([metricStateSecondaryMetricOptionsSchema, percentileRanksOperationSchema]),
-        schema.allOf([metricStateSecondaryMetricOptionsSchema, staticOperationDefinitionSchema]),
-        schema.allOf([metricStateSecondaryMetricOptionsSchema, uniqueCountMetricOperationSchema]),
+        // oneOf allows only 12 items
```

Comment:
extrac this so its not repeated (as else it will be repeated for all the charts)